diff --git a/libc-top-half/CMakeLists.txt b/libc-top-half/CMakeLists.txt
--- a/libc-top-half/CMakeLists.txt
+++ b/libc-top-half/CMakeLists.txt
@@ -538,7 +538,6 @@ add_internal_object_library(bulk-memory ${MUSL_BULK_MEMORY_SOURCES})
 
 foreach(obj bulk-memory-shared bulk-memory-static)
   target_compile_definitions(${obj} PRIVATE -DBULK_MEMORY_THRESHOLD=${BULK_MEMORY_THRESHOLD})
-  target_compile_options(${obj} PRIVATE -mbulk-memory)
 endforeach()
 
 # =============================================================================
